ocaml/xapi/db_gc.ml

```diff
@@ -77,45 +78,24 @@ let detect_clock_skew ~__context host skew =
 (* Master compares the database with the in-memory host heartbeat table and sets the live flag accordingly.
    Called with the use_host_heartbeat_for_liveness_m and use_host_heartbeat_for_liveness is true (ie non-HA mode) *)
 let check_host_liveness ~__context =
-  (* Check for rolling upgrade mode - if so, use host metrics for liveness else use hashtbl *)
-  let rum =
-    try Helpers.rolling_upgrade_in_progress ~__context with _ -> false
-  in
   (* CA-16351: when performing the initial GC pass on first boot there won't be a localhost *)
   let localhost = try Helpers.get_localhost ~__context with _ -> Ref.null in
-  (* Look for "true->false" transition on Host_metrics.live *)
   let check_host host =
     if host <> localhost then
       try
         let hmetric = Db.Host.get_metrics ~__context ~self:host in
         let live = Db.Host_metrics.get_live ~__context ~self:hmetric in
-        (* See if the host is using the new HB mechanism, if so we'll use that *)
-        let new_heartbeat_time =
+        let timer =
           with_lock host_table_m (fun () ->
-              Option.value
-                (Hashtbl.find_opt host_heartbeat_table host)
-                ~default:Clock.Date.(epoch |> to_unix_time)
+              match Hashtbl.find_opt host_heartbeat_table host with
+              | Some x ->
+                  x
+              | None ->
+                  Clock.Timer.start
+                    ~duration:!Xapi_globs.host_assumed_dead_interval
```

Comment:
I think in case of `None`, it means the host has never tickle heartbeat? Then we should consider it as expired?

Reply:
```
287 let start_db_gc_thread () =
288   Thread.create
289     (fun () ->
290       Debug.with_thread_named "db_gc"
291         (fun () ->
292           while true do
293             try Thread.delay db_GC_TIMER ; single_pass ()
294             with e ->
295               debug "Exception in DB GC thread: %s" (ExnHelper.string_of_exn e)
296           done
297         )
298         ()
299     )
300     ()
301
```

For
`293             try Thread.delay db_GC_TIMER ; single_pass ()`
As we use a timer to check heartbeat and a default timer for each host, I think here we should start `single_pass` immediately.